### PR TITLE
nwc: eom-harvesting & misc changes 

### PR DESCRIPTION
**Pull request by vivacebelles**
The primary focus of this PR will be activating the regex for EOM functionality as current energy harvesting isn't capable. EOM gradient was capable with a minor hiccup of the symmetry that can be reviewed in Issue #38. 

Additional edits are options edits for properties and updating nwc tests with new language in updating qcel and qcengine.

### CODE REVIEW

File: qcdb/tests/nwchem_tests/test_tce_ccd.py [hunk 1]
```diff
@@ -14,16 +14,16 @@ def tce_ccd(return_value):
...
 
     assert compare_values(hf, qcdb.variable('HF TOTAL ENERGY'), 5, 'hf ref')
     assert compare_values(ccd_tot, qcdb.variable('CCD TOTAL ENERGY'), 5, 'ccsd tot')
     assert compare_values(ccd_corl, qcdb.variable('CCD CORRELATION ENERGY'), 5, 'ccsd corl')
-    assert compare_values(x, qcdb.variable('CURRENT DIPOLE X'), 5, 'x dipole')
-    assert compare_values(y, qcdb.variable('CURRENT DIPOLE Y'), 5, 'y dipole')
-    assert compare_values(z, qcdb.variable('CURRENT DIPOLE Z'), 5, 'z dipole')
+    #assert compare_values(x, qcdb.variable('CURRENT DIPOLE X'), 5, 'x dipole')
+    #assert compare_values(y, qcdb.variable('CURRENT DIPOLE Y'), 5, 'y dipole')
+    #assert compare_values(z, qcdb.variable('CURRENT DIPOLE Z'), 5, 'z dipole')
```
Comment (loriab) [inline qcdb/tests/nwchem_tests/test_tce_ccd.py]: did you mean to stop testing dipole here?

File: qcdb/tests/nwchem_tests/test_tddft_n2+.py [hunk 2]
```diff
@@ -35,39 +34,41 @@ def check_tddft(return_value):
...
+        1 2 
         N 0.0 0.0 -0.54885
         N 0.0 0.0  0.54885
+        symmetry d2h
         ''')
 
     qcdb.set_options({
         'basis': '6-31g**',
         'memory' : '3000 mb',
+        'nwchem_charge' : 1,
```
Comment (loriab) [inline qcdb/tests/nwchem_tests/test_tddft_n2+.py]: this is redundant, right?

Comment (vivacebelles) [reply qcdb/tests/nwchem_tests/test_tddft_n2+.py]: Yes, I'll remove that in the next set of updates.

File: qcdb/qcvars/glossary.py [hunk 3]
```diff
@@ -756,6 +756,19 @@ def define_ex_transition_qcvars(max_root, mtd, sym):
...
+def define_tddft_roots_qcvars(max_root, sym):
+    sym = []
+
+    for i in range(max_root):
+        qcvardefs['TDDFT ROOT {} EXCITATION ENERGY - {} SYMMETRY'.format(i+1, sym)] = {
+                'units' : 'Eh',
+                'glossary' : """ The excitation energy of time-dependent DFT in {} symmetry from 0 to root {}""".format(sym, i+1)
+                }
+
+        qcvardefs['TDDFT ROOT {} EXCITED STATE ENERGY - {} SYMMETRY'.format(i+1, sym)] = {
```
Comment (loriab) [inline qcdb/qcvars/glossary.py]: this should be `EXCITED STATE` --> `TOTAL` and the glossary entry edited, right? Something like "The total energy of the {}-th excited state (GS is root 0) by time-dependent DFT of symmetry {}"

File: qcdb/qcvars/glossary.py [hunk 4]
```diff
@@ -1820,6 +1833,80 @@ def define_ex_transition_qcvars(max_root, mtd, sym):
...
+    'Eh',
+    'glossary':
+    r"""
+The number of alpha electrons
+"""
+}
+
+qcvardefs['N BETA ELECTRONS'] = {
+    'units':
+    'Eh',
```
Comment (loriab) [inline qcdb/qcvars/glossary.py]: unitless

File: qcdb/tests/nwchem_tests/test_prop_dipole_h2o.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+import numpy as np
+#import pprinter as pp
+
+def check_dipole(return_value):
+
+    dip_x = -0.0000000000
+    dip_y = -0.0000000000
+    dip_z = -2.2247923027
+    dip_grad = np.array([-0.0000000, -0.000000, -2.2247923027])
```
Comment (loriab) [inline qcdb/tests/nwchem_tests/test_prop_dipole_h2o.py]: this is dipole in array format (3,), not dipole gradient (3*nat, 3), right?

**Pull request merged**